sdk/python/lib/pulumi/runtime/resource.py

```diff
@@ -911,6 +921,33 @@ def _create_custom_timeouts(
     return result
 
 
+def _struct_value_to_python(value: struct_pb2.Value) -> Any:
+    """
+    Converts a struct_pb2.Value to Python native types recursively.
+    This is used when we receive a struct_pb2.Value from transforms and need to re-serialize it.
+    """
+    kind = value.WhichOneof("kind")
+    if kind == "null_value":
+        return None
+    elif kind == "number_value":
+        return value.number_value
+    elif kind == "string_value":
+        return value.string_value
+    elif kind == "bool_value":
+        return value.bool_value
+    elif kind == "struct_value":
+        # Recursively convert all nested struct_pb2.Value objects
+        result = {}
+        for k, v in value.struct_value.fields.items():
+            result[k] = _struct_value_to_python(v)
+        return result
```

Comment:
If we are going to use list comprehension for lists, we should use dictionary comprehension for "value"s. 
```suggestion
        return {k: _struct_value_to_python(v) for k, v in value.struct_value.fields.items()}
```